### PR TITLE
CAKE-5292 | Use only one call to the service AND OTHER HOTFIXES

### PR DESCRIPTION
**Pull request by vforge**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5292

## Description

- Rework how `affiliate-slot` service works, always call service once and always return `{big: unit, small: unit}` object from Promise
- other hotfixes

## Reviewers

@Wikia/cake

### CODE REVIEW

File: app/components/article-content.js [hunk 1]
```diff
@@ -605,92 +604,76 @@ export default Component.extend(
...
+            } else {
+              const unitPlaceholder = document.createElement('div');
+              const unitWrapper = document.createElement('div');
+              unitWrapper.appendChild(unitPlaceholder);
+              h2Elements[indexForBigUnit].insertAdjacentElement('beforebegin', unitWrapper);
+
+              this.renderedComponents.push(this.renderComponent({
+                name: 'affiliate-unit',
+                attrs: {
+                  unit: bigUnit,
```
Comment (cschucode) [inline app/components/article-content.js:641]: Just a question, but were setting the placeholder's unit attribute to always `undefined` here?

Comment (vforge) [reply app/components/article-content.js:641]: No, why?

Comment (JoshuaRogan) [reply app/components/article-content.js:641]: I think you are reading the wrong if statement.

**Comment by JoshuaRogan** [comment]
New debug feat: 
`debugAffiliateUnits=disneyplus,disney,true` the third param is ignoreBigUnit which can be ignored or specified to only display a small unit.

**Review by JoshuaRogan — APPROVED** [review]

File: app/services/affiliate-slots-units.json [hunk 2]
```diff
@@ -112,5 +112,167 @@
...
+    "isBig": true,
+    "launchOn": "2019-11-18T22:00:00Z"
+  },
+  {
+    "campaign": "disneyplus",
+    "category": "marvel",
+    "image": "https://static.wikia.nocookie.net/d8374c50-15b4-4ee6-97be-fa8dc4d5976e",
+    "heading": "All Avengers, All the Time",
+    "subheading": "Watch Now On Disney+",
+    "link": "https://disneyplus.bn5x.net/beeqP",
```
Comment (mklucsarits) [inline app/services/affiliate-slots-units.json]: This is the old link

File: app/services/affiliate-slots-units.json [hunk 3]
```diff
@@ -112,5 +112,167 @@
...
+    ],
+    "launchOn": "2019-11-18T22:00:00Z"
+  },
+  {
+    "campaign": "disneyplus",
+    "category": "marvel",
+    "image": "https://static.wikia.nocookie.net/d8374c50-15b4-4ee6-97be-fa8dc4d5976e",
+    "heading": "All Avengers, All the Time",
+    "subheading": "Watch Now On Disney+",
+    "link": "https://disneyplus.bn5x.net/beeqP",
```
Comment (mklucsarits) [inline app/services/affiliate-slots-units.json]: This also needs to be updated

File: app/components/article-content.js (hunk 1 above)
Comment (cschucode) [reply app/components/article-content.js:641]: See it now. if inside the if. Thanks. 👍

File: app/services/affiliate-slots-units.json (hunk 3 above)
Comment (JoshuaRogan) [reply app/services/affiliate-slots-units.json]: Done https://github.com/Wikia/mobile-wiki/pull/1913/commits/ab61b542efc808b2bd2768eeef6c29dce3977dbe

File: app/services/affiliate-slots-units.json (hunk 2 above)
Comment (JoshuaRogan) [reply app/services/affiliate-slots-units.json]: https://github.com/Wikia/mobile-wiki/pull/1913/commits/ab61b542efc808b2bd2768eeef6c29dce3977dbe

**Review by mklucsarits — APPROVED** [review]

**Pull request merged**